Remove improperly applied flattening extension.

jhendrixMSFT opened the pull request:
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] Validation errors from the [Linter extension for VS Code](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/linter.md) have all been fixed for this spec. (**Note:** for large, previously checked in specs, there will likely be many errors shown. Please contact our team so we can set a timeframe for fixing these errors if your PR is not going to address them).
  - [ ] The spec follows the patterns described in the [Swagger good patterns](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-good-patterns.md) document unless the service API makes this impossible.
The x-ms-client-flatten extension has been applied in the wrong location.
Unfortunately fixing it would introduce a breaking change so instead we
will just remove the extension.

azurecla commented:
Hi __@jhendrixMSFT__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (jhendrix). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

jhendrixMSFT commented:
@amarzavery Here's the PR for the issue we were discussing earlier.

amarzavery commented:
Although this is fixing the application of the client-flattening extension in a correct way, this will cause a breaking change in the DeploymentProperties model class. It will no longer have a property named debugSetting of type "#/definitions/DebugSetting". Instead the class will have a property named "detailLevel" of type "string". 

At the CLI/Powershell level the commands to validate deployment and create/update deployment will have to make changes in the command implementation. I dont think there would be any changes in the API of these commands.

The SDK would have breaking changes in the structure of the body parameter (Deployment) being consumed by Deployments_validate and Deployments_CreateOrUpdate operations.

If everyone is ok with this change then feel free to merge the PR.

@jhendrixMSFT, @veronicagg, @lmazuel @yugangw-msft, @jianghaolu, @johnaste, @markcowl

vivsriaus commented:
@jhendrixMSFT can you please let us know what is the issue with the current implementation? The current one actually mirrors the service side model - more amenable if we do introduce future properties to debug setting.

jhendrixMSFT commented:
@vivsriaus The property of DebugSettings is a primary type so there's nothing to flatten.  Are you saying this is intentional?

amarzavery commented on arm-resources/resources/2016-09-01/swagger/resources.json:
```diff
@@ -1853,6 +1853,7 @@
           }
         },
         "debugSetting": {
+          "x-ms-client-flatten": true,
```
@jhendrixMSFT - I agree that the flattening extension was previously applied at an incorrect place and this change is trying to apply it correctly. However, this will induce a breaking change in the sdk. This can affect our customers. The sdks have been shipped like this since a long time. I believe customers are used to accessing this as `debugSetting.detailLevel`.

So the right thing would be to
- remove this flattening extension that you have applied correctly
- remove the incorrectly applied flattening extension

Moral of the story: We wont flatten here. This will ensure everything is business as usual. Let me know if my thoughts are not clear.
Note: Have spoken with @vivsriaus and he is in agreement with the recommendation.

jhendrixMSFT replied:
I'm fine with removing the extension and can update this PR if that is the recommendation.

amarzavery replied:
@jhendrixMSFT - Can you please update the PR by removing the x-ms-client-flatten extension? Once that is done, the PR can be merged.

jhendrixMSFT committed:
Remove improperly applied flattening extension.

The x-ms-client-flatten extension has been applied in the wrong location.
Unfortunately fixing it would introduce a breaking change so instead we
will just remove the extension.

jhendrixMSFT commented:
@amarzavery PR updated.

AutorestCI commented:
[![Build Status](https://travis-ci.org/AutorestCI/azure-sdk-for-ruby.svg?branch=RestAPI-PR763)](https://travis-ci.org/AutorestCI/azure-sdk-for-ruby) https://github.com/Azure/azure-sdk-for-ruby/pull/645

AutorestCI commented:
No modification for NodeJS

AutorestCI commented:
[![Build Status](https://travis-ci.org/AutorestCI/azure-sdk-for-python.svg?branch=RestAPI-PR763)](https://travis-ci.org/AutorestCI/azure-sdk-for-python) https://github.com/Azure/azure-sdk-for-python/pull/949
